src/charmlibs/pathops/_container_path.py

```diff
@@ -93,37 +149,59 @@ def _can_compare(self, other: object) -> bool:
         return isinstance(other, ContainerPath) and other._container.name == self._container.name
 
     def __eq__(self, other: object, /) -> bool:
+        """Compare paths if other is a ContainerPath on the same container, else return False.
+
+        To explicitly compare paths, consider using :class:`str` to get the canonical path string,
+        or create a :class:`ContainerPath` with the same container using :meth:`with_segments`.
+        """
         if not isinstance(other, ContainerPath) or self._container.name != other._container.name:
             return False
         return self._path == other._path
 
     def __truediv__(self, key: StrPathLike) -> Self:
         """Return a new ContainerPath with the same container and the joined path.
 
-        The joined path is equivalent to str(self) / pathlib.PurePath(key).
+        The joined path is equivalent to ``str(self) / pathlib.PurePath(key)``.
 
-        Note that the right hand operand here cannot be a ContainerPath.
-        This is because ContainerPath objects only support absolute paths currently,
-        so using one as the right hand side is likely to be an error.
-        For the same reason, __rtruediv__ is undefined, meaning a ContainerPath cannot
-        be the right hand side operand for arbitrary string or PathLike objects either.
+        .. warning::
+            ``__rtruediv__`` is not supported, as :class:`ContainerPath` only supports absolute
+            paths. Since an absolute path as the right hand operand will completely replace the
+            left hand path, this is mostly likely to be a user error, so it isn't supported.
         """
         return self.with_segments(self._path, key)
 
     def is_absolute(self) -> bool:
-        """Whether the path is absolute. Will always be True, as relative paths error on init."""
+        """Return True if the path is absolute (has a root), which is always the case.
+
+        .. note::
+            Always ``True``, since initialising a :class:`ContainerPath` with a non-absolute
+            path will result in a :class:`RelativePathError`.
```

Comment:
I'd probably simplify to:

```suggestion
       Initialising a :class:`ContainerPath` with a non-absolute
       path will result in a :class:`RelativePathError`.
```

Reply:
I agree - what you have now (below) looks better than what I suggested.

```
"""Return whether the path is absolute (has a root), which is always the case.

Always ``True``, since initialising a :class:`ContainerPath` with a non-absolute
path will result in a :class:`RelativePathError`.
"""
```